lib/coverband/adapters/hash_redis_store.rb

```diff
@@ -0,0 +1,164 @@
+# frozen_string_literal: true
+
+module Coverband
+  module Adapters
+    class HashRedisStore < Base
+      FILE_KEY = 'file'
+      FILE_LENGTH_KEY = 'file_length'
+      META_DATA_KEYS = [DATA_KEY, FIRST_UPDATED_KEY, LAST_UPDATED_KEY].freeze
+      ###
+      # This key isn't related to the coverband version, but to the interal format
+      # used to store data to redis. It is changed only when breaking changes to our
+      # redis format are required.
+      ###
+      REDIS_STORAGE_FORMAT_VERSION = 'coverband_3_3'
+
+      attr_reader :redis_namespace
+
+      def initialize(redis, opts = {})
+        super()
+        @redis_namespace = opts[:redis_namespace]
+        @format_version = REDIS_STORAGE_FORMAT_VERSION
+        @redis = redis
```

Comment:
Do we need to do a check on minimum version supported or anything else to ensure the lua script will work?

Reply:
Done!